### DISCUSSION TITLE
bagv2/etl/etl.sh runnen

### DISCUSSION THREAD

**Discussion by anoukev** [nlextract/NLExtract#380] [Q&A] [answered]
Allereerst, ik heb nog nooit eerder zo'n vraag gesteld, mijn excuses als mijn vraag niet helemaal aan gebruikelijke conventies voldoet. Ik probeer NLExtract te gebruiken om de bag op te halen, maar loop tegen wat problemen aan. Ik hoop dat iemand mij hierbij kan helpen.

Ik heb een vraag over het runnen van bagv2/etl/etl.sh. Ik probeer de bag op te halen via het runnen van etl.sh, maar ik krijg tijdens het runnen een aantal errors. 
Ik krijg: 
`AttributeError: 'ConfigParser' object has no attribute 'readfp'. Did you mean: 'read'?`
Die veroorzaakt wordt door deze regel in util.py: `cp.readfp(FakeSecHead(open(file_path))) `

Als ik dit aanpas naar read, krijg ik `KeyError: 'asection`
Die veroorzaakt wordt door deze regel in util.py: `return cp._sections['asection']`

Ik vroeg mij af wat ik hier mee kan doen en hoe ik dit kan oplossen. 
Dit is mijn workflow:
Ik werk in een ubuntu omgeving (Ubuntu 22.04.3 LTS) en ga de volgende commands langs:
- `wget https://service.pdok.nl/lv/bag/atom/downloads/lvbag-extract-nl.zip`
- `wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh`
- `bash Miniconda3-latest-Linux-x86_64.sh`
- `conda create --name bag2`
- `conda activate bag2`
- `conda install -c conda-forge gdal`
- `conda update --all`
- `conda upgrade -c conda-forge gdal`
-  `wget https://github.com/nlextract/NLExtract/files/10372701/nlextract-1.5.4.zip`
- `unzip nlextract-1.5.4.zip`
- `pip install deprecated`
- `pip install psycopg2-binary`
- `pip install lxml`
- `conda activate bag2`
- `cd nlextract-1.5.4/bagv2/etl/`
- ` ./etl.sh -a schema=xxx -a bag_input_file=../../../lvbag-extract-nl.zip -a pg_host=localhost -a pg_port=xxx -a pg_user=postgres -a password=xxx -a pg_db=DB`

na het runnen van dit laatste command krijg ik de bovengenoemde errors. Deze komen ook als ik alleen `./etl.sh` run.
Dit zijn de packages in mijn environment:
Deprecated==1.2.14
GDAL @ file:///home/conda/feedstock_root/build_artifacts/gdal-split_1711361891382/work/build/swig/python
lxml==5.1.0
numpy @ file:///croot/numpy_and_numpy_base_1708638617955/work/dist/numpy-1.26.4-cp312-cp312-linux_x86_64.whl#sha256=b8e59fcb10ab071cbeded47aed247ddfa1c902d595848c27f82021340e226134
psycopg2-binary==2.9.9
setuptools==69.2.0
wheel==0.43.0
wrapt==1.16.0

**Comment by justb4** [comment, marked as answer]
Denk dat je een te hoge Python versie gebruikt bijv 3.12: https://bugs.launchpad.net/lazr.restfulclient/+bug/2041407 . Als het lukt lagere versie zoals 3.9.

**Reply by justb4** [reply]
En, gelukt?

**Reply by anoukev** [reply]
Het leek hierdoor wel beter te gaan, maar we hebben er toch voor gekozen om via geotoko de data te verkrijgen. Nog bedankt voor de hulp. Wat mij betreft kan deze discussie gesloten worden.
